File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l,r,res = 0,len(heights)-1,0
        while l < r:
            cur = min(heights[l],heights[r]) * (r-l)
            res = max(res,cur)
            if heights[l] < heights[r]:
                l+=1;
            else:
                r-=1;
        return res

File: Data Structures & Algorithms/partition-labels/submission-1.cpp
class Solution {
public:
    unordered_map<char,int> col;
    vector<int> partitionLabels(string s) {
        for (int i = 0; i < s.size(); i++){
            col[s[i]]=i;
        }
        vector<int> res;
        int start = 0;
        int farthest = 0;
        for (int i = 0; i < s.size(); i++) {
            farthest = max(farthest,col[s[i]]);
            if (i == farthest) {
                res.push_back(i-start+1);
                start = i+1;
            }
        }
        return res;
    }
};
